workspace.dsl

```diff
@@ -0,0 +1,167 @@
+workspace "Appeal service" {
+	model {
+		properties {
+			"structurizr.groupSeparator" "/"
+		}
+
+		userLocalPlanningAuthority = person "Local Planning Authority" "A local authority issuing decisions on planning applications"
+		userAppellant = person "Appellant" "A member of the public appealing against a planning decision"
+
+		group "Planning Inspectorate" {
+			userCaseOfficer = person "Case officer" "Checks LPA questionnaire and uploads statements and comments"
+			userInspector = person "Inspector" "Visits appeal site and issues appeal decision"
+
+			systemAppsFo = softwareSystem "Appeals Front-Office" "Public dashboards for appellants and Local Planning Authorities" "Node.js, Azure Web App, Azure Function App" {
+				tags "InternalCollaborationSystem"
+
+				containerFoFileStorage = container "File storage" "Stores all files uploaded in the Front-Office"
+			}
+
+			systemAppsBo = softwareSystem "Appeals Back-Office" "Internally facing service allowing for management of planning appeals in England"  {
+				tags "InternalCollaborationSystem"
+
+				group "Web" {
+
+    				containerBoWeb = container "Back-Office Web" "Back-office website" "Node.js, Azure Web App" {
+    					tags "Microsoft Azure - App Services - Web"
+    				}
+
+    				containerBoRedis = container "Redis" "Distributed session storage"
+    				containerBoAzureActiveDirectory = container "Entra ID" "Stores login details and assigned user groups"
+    				containerBoFileStorage = container "File storage" "Stores all files related to appeals"
+
+				}
+
+				group "API" {
+    				containerBoApi = container "Back-Office API" "" "Node.js, Azure Web App" {
+    					tags "Microsoft Azure - App Services - API"
+    					componentBoStateMachine = component "State machine" "Handles the appeal lifecycle and state transitions"
+    				}
+
+    				containerBoAzureSql = container "Database" "Data store for appeals" "Azure SQL"
+				}
+
+				group "Azure functions" {
+				    containerFunctionAppCasedata = container "Casedata import" "Consumes appellant and LPA submissions from Service Bus" "Function App, JavaScript"{
+						tags "Microsoft Azure - Function Apps", "FunctionApp"
+					}
+					containerFunctionAppDocs = container "Document processing" "Copies blobs from Front-Office, processes virus scanning" "Function App, JavaScript"{
+						tags "Microsoft Azure - Function Apps", "FunctionApp"
+					}
+				}
+			}
+
+			systemIntegration = softwareSystem "Integration Layer" "Event-driven message exchange between applications" {
```

Comment:
There's a few different ways to approach this, I'd recommend watching the following snippet which gives advice here https://youtu.be/x2-rSnhpw0g?feature=shared&t=1634

Also on https://c4model.com/ down the bottom of the page it notes

> Should you include message buses, API gateways, service meshes, etc?
If you have two services, A and B, that communicate by sending a message via a message bus (irrespective of topics, queues, p2p, pub/sub, etc) or another intermediary (e.g. an API gateway or service mesh), you have a couple of options. The first option is to show service A sending a message to the intermediary, and the intermediary subsequently forwarding that message to service B. While accurate, the "hub and spoke" nature of the diagram tends to obscure the notion that there's coupling between the message producer and consumer.
> 
> The other approach is to omit the intermediary, and instead use notation (e.g. a textual description, colour coding, line style, etc) to signify that the interaction between service A and B happens via an intermediary. This approach tends to result in diagrams that tell a clearer story.

Reply:
I'm curious to hear peoples thoughts on this one (@bryn-pins @beejjacobs), as there's no single right way of doing this. But I think we need to aim for what is most readable and understandable.

You can see the approach used for the Applications FO at https://planning-inspectorate.github.io/applications-service/master/applications-front-office/container/